docs: Add Developer Resources documentation to Documentation menu

aThorp96 opened the pull request:
# Changes <!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

Enable users to access the Developer Resources documentation (renamed Contributor Resources) from the side-menu on the site

# Submitter Checklist

- [x] 📝 Ensure your commit message is clear and informative. Refer to the How to write a git commit message guide. Include the commit message in the PR body rather than linking to an external site (e.g., Jira ticket).

- [x] ♽ Run make test lint before submitting a PR to avoid unnecessary CI processing. Consider installing [pre-commit](https://pre-commit.com/) and running pre-commit install in the repository root for an efficient workflow.

- [x] ✨ We use linters to maintain clean and consistent code. Run make lint before submitting a PR. Some linters offer a --fix mode, executable with make fix-linters (ensure [markdownlint](https://github.com/DavidAnson/markdownlint) and [golangci-lint](https://github.com/golangci/golangci-lint) are installed).

- [ ] 📖 Document any user-facing features or changes in behavior.

- [ ] 🧪 While 100% coverage isn't required, we encourage unit tests for code changes where possible.

- [ ] 🎁 If feasible, add an end-to-end test. See [README](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/README.md) for details.

- [x] 🔎 Address any CI test flakiness before merging, or provide a valid reason to bypass it (e.g., token rate limitations).

- If adding a provider feature, fill in the following details:

| Git Provider     | Supported |
|------------------|-----------|
| GitHub App       | ✅️        |
| GitHub Webhook   | ❌️        |
| Gitea            | ❌️        |
| GitLab           | ❌️        |
| Bitbucket Cloud  | ❌️        |
| Bitbucket Server | ❌️        |

  (update the documentation accordingly)

aThorp96 committed:
docs: Add Developer Resources documentation to Documentation menu

The Developer Resources page was previously inaccessible via the
static site except through the site's search results.

enarha commented:
/lgtm

chmouel commented:
SGTM any chance you can add that link to the contributing section of the README as well https://github.com/openshift-pipelines/pipelines-as-code/tree/main?tab=readme-ov-file#contributing

chmouel commented:
and hack/dev/kind too 

![image](https://github.com/user-attachments/assets/12876208-c695-484e-b7f1-16c4c574fdb5)

aThorp96 committed:
docs: Rename "Developer Resources" section to "Contributor Resources"

Since CI/CD is often written and maintained by developers, using the
term "Contributor" more-clearly distinguishes the target of the
developer docs, since they are targetted at developers attempting
to contribute to PaC

aThorp96 commented:
Good catch @chmouel, thanks. Done

That does make me think: I wonder if markdownlint could either detect dead links in a PR and/or detect dead links to internal documentation 🤔 Out of scope for this PR of course but the thought is rattling around in my brain now 😄

chmouel commented:
just a note we prefer squashed commits (in tektoncd/ as well) it's fine now but next time please squash the commits unless it's distinctive features/commits that are not related

aThorp96 commented:
Thanks @chmouel! For squash commits, do we generally prefer one commit per PR or to squash into atomic changes?

mathur07 commented:
@aThorp96 [https://pipelinesascode.com/docs/dev/](https://pipelinesascode.com/docs/dev/) is 404 not found can you take a look:
![image](https://github.com/user-attachments/assets/08fac528-86a8-4753-8d2b-5ba94ae47fff)

aThorp96 commented:
@mathur07 I was just asking Chmouel about this lol

This change is merged but the main site only gets updated when a new release is cut. You can find "bleeding edge" docs on https://main.pipelines-as-code.pages.dev/docs/dev/ or https://nightly.pipelines-as-code.pages.dev/docs/dev/

chmouel commented:
ah yeah there is as well: 


https://docs.pipelinesascode.com/

but i never got into making it generating nightly or promot that url

> On 11 Feb 2025, at 19:14, Andrew Thorp ***@***.***> wrote:
> 
> 
> @mathur07 <https://github.com/mathur07> I was just asking Chmouel about this lol
> 
> This change is merged but the main site only gets updated when a new release is cut. You can find "bleeding edge" docs on https://main.pipelines-as-code.pages.dev/docs/dev/ or https://nightly.pipelines-as-code.pages.dev/docs/dev/
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/openshift-pipelines/pipelines-as-code/pull/1900#issuecomment-2651636433>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAYFJDMTDEIJD2XWKQWV7D2PI4X5AVCNFSM6AAAAABWIDYG6OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNJRGYZTMNBTGM>.
> You are receiving this because you were mentioned.
>

mathur07 commented:
> @mathur07 I was just asking Chmouel about this lol
> 
> This change is merged but the main site only gets updated when a new release is cut. You can find "bleeding edge" docs on https://main.pipelines-as-code.pages.dev/docs/dev/ or https://nightly.pipelines-as-code.pages.dev/docs/dev/

Thanks!
